src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -340,4 +364,23 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		}
 		return undefined;
 	}
+
+	public fileToBase64 = (filepath, getType) => {
+		return new Promise<string | ArrayBuffer>(async resolve => {
+			let response = await fetch(filepath);
```

Comment:
Is there a reason you're using fetch here instead of just using the file service?

Reply:
yeah, our file service doesn't return the base64 value, it returns IFileContent (VSBuffer value) and I couldn't find a way to convert that to a blob.

Reply:
Why do you need a blob? See the other comment I left about using btoa/atob

Reply:
I actually tried using that first, atob takes in a string value and I tried passing the filecontents read using file service and I was not able to get a valid base64 value. (Am not sure what's string value of an image file in this case) So I searched for alternate solutions.

Reply:
A file is a file - you should be able to get the string contents regardless of what the type is. What difficulties were you having?

For example here's a version using fs : https://gist.github.com/crspiccin/790796a68e7178404de4 which just reads it as base64 directly.

(we can't use fs directly so you should use the FileService, but the concept should be the same)

Reply:
So the below is the error that I got when I used FileService and btao:
Error: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.

Reply:
Did you see https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/browser/dom.ts#L1479?